### PR TITLE
refactor: simplify `proof-of-sql-parser`

### PR DESCRIPTION
**Pull request by iajoiner**
# Rationale for this change
For intermediate AST -> Provable AST and intermediate AST postprocessing evaluation tests in `proof-of-sql` it is a good idea to be able to use shortcuts in `proof_of_sql_parser::test_utility`. 

# What changes are included in this PR?
- put `test_utility.rs` in parser behind a feature flag and allow it to be used by the other crate
- prevent `posql_time` from being unnecessarily convoluted

# Are these changes tested?
Existing tests should pass

### CODE REVIEW

**Review by drcapybara — DISMISSED** [review]
This is a nice refactor, thanks. Feel free to move the ```Timestamp``` error module too if it makes sense

**Review by drcapybara — APPROVED** [review]
Looks nice, thanks

File: crates/proof-of-sql-parser/src/lib.rs [hunk 1]
```diff
@@ -9,11 +9,12 @@ extern crate lalrpop_util;
 
 pub mod intermediate_ast;
 
-#[cfg(test)]
+#[cfg(all(test, feature = "parser-test-utility"))]
```
Comment (JayWhite2357) [inline crates/proof-of-sql-parser/src/lib.rs:12]: Is this change needed? These are just tests, right?
```suggestion
#[cfg(test)]
```

File: crates/proof-of-sql-parser/src/test_utility.rs
Comment (JayWhite2357) [inline crates/proof-of-sql-parser/src/test_utility.rs:1]: Any reason we can't simply make this available by default? I don't see a problem with calling this `utility` instead of `test_utility` and just simply making it `pub` instead of putting it behind a feature.

My only suggestion in that case would be to make the documentation a bit more verbose.

**Review by JayWhite2357 — APPROVED** [review]
Approved if you don't want to apply my suggestions.

**Comment by iajoiner** [comment]
@JayWhite2357 Actually I will..in a future PR very soon.
